src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/ReadHelper.cs

```diff
@@ -378,6 +383,59 @@ FROM entities
         return result;
     }
 
+    public async Task<Dictionary<NonFungibleGlobalIdLookup, NonFungibleIdGlobalIdDatabaseId>> ReadNonFungibleData(
+        HashSet<NonFungibleGlobalIdLookup> referencedNonFungibleIds,
+        CancellationToken token)
+    {
+        if (!referencedNonFungibleIds.Any())
+        {
+            return new Dictionary<NonFungibleGlobalIdLookup, NonFungibleIdGlobalIdDatabaseId>();
+        }
+
+        var sw = Stopwatch.GetTimestamp();
+
+        var resourceAddresses = new List<string>();
+        var nonFungibleIds = new List<string>();
+
+        foreach (var nf in referencedNonFungibleIds)
+        {
+            resourceAddresses.Add(nf.ResourceAddress);
+            nonFungibleIds.Add(nf.SimpleRepresentation);
+        }
+
+        var cd = new CommandDefinition(
+            commandText: $@"
+WITH var (non_fungible_resource_address, non_fungible_id) AS (
+    SELECT UNNEST(@resourceAddresses), UNNEST(@nonFungibleIds)
+)
+SELECT
+    var.non_fungible_resource_address AS ResourceAddress,
+    var.non_fungible_id AS SimpleRepresentation,
+    nfid.non_fungible_resource_entity_id AS ResourceEntityId,
+    nfid.id AS NonFungibleIdDataId
+FROM var
+INNER JOIN entities e ON e.address = var.non_fungible_resource_address
+INNER JOIN LATERAL (
```

Comment:
`non_fungible_id_data` does not contain history entries. It should be OK to just use regular INNER JOIN.